Lib/fontTools/cu2qu/ufo.py

```diff
@@ -143,22 +143,25 @@ def _set_segments(glyph, segments, reverse_direction):
             raise AssertionError('Unhandled segment type "%s"' % tag)
 
 
-def _segments_to_quadratic(segments, max_err, stats):
+def _segments_to_quadratic(segments, max_err, stats, all_quadratic=True):
     """Return quadratic approximations of cubic segments."""
 
     assert all(s[0] == "curve" for s in segments), "Non-cubic given to convert"
 
-    new_points = curves_to_quadratic([s[1] for s in segments], max_err)
+    new_points = curves_to_quadratic([s[1] for s in segments], max_err, all_quadratic)
     n = len(new_points[0])
     assert all(len(s) == n for s in new_points[1:]), "Converted incompatibly"
 
     spline_length = str(n - 2)
     stats[spline_length] = stats.get(spline_length, 0) + 1
 
-    return [("qcurve", p) for p in new_points]
+    if all_quadratic:
+        return [("qcurve", p) for p in new_points]
+    else:
+        return [("qcurve" if len(p) == 3 else "curve", p) for p in new_points]
```

Comment:
I'm not 100% sure this line is correct. The `_segments_to_quadratic` is called when zipping through on each curves at ith position along the interpolatable contours. I expect its return value to be all qcurve or all curve, not a mix of the two, otherwise it'd break interpolability.
Maybe that's impossible because `curves_to_quadratic` already does the right thing, in which case this line could be further simplified because we already know `n = len(new_points[0])` a few lines above, if that's == 3 and all_quadratic=False, then we know (or assert) that the segment type is qcurve, otherwise curve... Does that make sense?

Reply:
Yes, I believe it can be simplified as you said.